subworkflows/ebi-metagenomics/amr_annotation/main.nf

```diff
@@ -0,0 +1,233 @@
+            ])
+            WGET_CARD(ch_card_url)
+            ch_versions = ch_versions.mix(WGET_CARD.out.versions.first())
+            UNTAR_CARD(WGET_CARD.out.outfile)
+            ch_versions = ch_versions.mix(UNTAR_CARD.out.versions.first())
+            rgi_db = UNTAR_CARD.out.untar.map { it[1] }
+            RGI_CARDANNOTATION(rgi_db)
+            card = RGI_CARDANNOTATION.out.db
+            ch_versions = ch_versions.mix(RGI_CARDANNOTATION.out.versions.first())
+        }
+        else {
+
+            // Use user-supplied database
+            rgi_db = file(params.arg_rgi_db, checkIfExists: true)
+            if (!rgi_db.contains("card_database_processed")) {
+                RGI_CARDANNOTATION(rgi_db)
+                card = RGI_CARDANNOTATION.out.db
+                ch_versions = ch_versions.mix(RGI_CARDANNOTATION.out.versions.first())
+            }
+            else {
+                card = rgi_db
+            }
+        }
+
+        RGI_MAIN(ch_faa, card, [])
+        ch_versions = ch_versions.mix(RGI_MAIN.out.versions.first())
+
+        // Reporting
+        HAMRONIZATION_RGI(RGI_MAIN.out.tsv, 'tsv', RGI_MAIN.out.tool_version, RGI_MAIN.out.db_version)
+        ch_versions = ch_versions.mix(HAMRONIZATION_RGI.out.versions.first())
+        ch_rgi_results = HAMRONIZATION_RGI.out.tsv
+    }
+
+    // DeepARG prepare download
+    if (!params.arg_skip_deeparg && params.arg_deeparg_db) {
+        ch_deeparg_db = channel
+            .fromPath(params.arg_deeparg_db, checkIfExists: true)
+            .first()
+    }
+    else if (!params.arg_skip_deeparg && !params.arg_deeparg_db) {
+        ch_deeparg_url = channel.of([
+            [ id: 'deeparg_db' ],
+            'https://zenodo.org/records/8280582/files/deeparg.zip?download=1'
+        ])
+    
+        // Download the database using WGET
+        WGET_DEEPARG( ch_deeparg_url )
+        ch_versions = ch_versions.mix(WGET_DEEPARG.out.versions.first())
+    
+        // Unzip the downloaded database
+        UNZIP_DEEPARG(WGET_DEEPARG.out.outfile )
+        ch_versions = ch_versions.mix(UNZIP_DEEPARG.out.versions.first())
+    
+        // Extract the unzipped directory path
+        ch_deeparg_db = UNZIP_DEEPARG.out.unzipped_archive.map { meta, dir -> dir }
+    }
+
+    // DeepARG run
+    if (!params.arg_skip_deeparg) {
+        ch_faa
+            .map { it ->
+                def meta = it[0]
+                def anno = it[1]
+                def model = params.arg_deeparg_model
+                [meta, anno, model]
+            }
+            .set { ch_input_for_deeparg }
+
+        DEEPARG_PREDICT(ch_input_for_deeparg, ch_deeparg_db)
+        ch_versions = ch_versions.mix(DEEPARG_PREDICT.out.versions.first())
+
+        // Reporting
+        // Note: currently hardcoding versions as unreported by DeepARG
+        // Make sure to update on version bump
+        HAMRONIZATION_DEEPARG(DEEPARG_PREDICT.out.arg, 'tsv', '1.0.4', params.arg_deeparg_db_version)
```

Comment:
Maybe have this as an input... so we don't have to update the module if we change the DB